python_modules/libraries/dagster-github/dagster_github/resources.py

```diff
@@ -0,0 +1,176 @@
+import time
+from datetime import datetime
+from dagster import resource, Field, String, Int
+import requests
+import jwt
+import pem
+
+
+def to_seconds(dt):
+    return (dt - datetime(1970, 1, 1)).total_seconds()
+
+
+class GithubResource:
+    def __init__(self, client, app_id, app_private_rsa_key, default_installation_id):
+        self.client = client
+        self.app_private_rsa_key = app_private_rsa_key
+        self.app_id = app_id
+        self.default_installation_id = default_installation_id
+        self.installation_tokens = {}
+        self.app_token = {}
+
+    def __set_app_token(self):
+        # from https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/
+        # needing to self-sign a JWT
+        now = int(time.time())
+        # JWT expiration time (10 minute maximum)
+        expires = now + (10 * 60)
+        certs = pem.parse(str.encode(self.app_private_rsa_key))
+        if len(certs) == 0:
+            raise Exception("Invalid app_private_rsa_key")
+        encoded_token = jwt.encode(
+            {
+                # issued at time
+                "iat": now,
+                # JWT expiration time
+                "exp": expires,
+                # GitHub App's identifier
+                "iss": self.app_id,
+            },
+            str(certs[0]),
+            algorithm="RS256",
+        ).decode("utf-8")
+        self.app_token = {
+            "value": encoded_token,
+            "expires": expires,
+        }
+
+    def __check_app_token(self):
+        print("check app token")
+        if ("expires" not in self.app_token) or (
+            self.app_token["expires"] < (int(time.time()) + 60)
+        ):
+            self.__set_app_token()
+
+    def get_installations(self, headers={}):
+        self.__check_app_token()
+        headers["Authorization"] = "Bearer {}".format(self.app_token["value"])
+        headers["Accept"] = "application/vnd.github.machine-man-preview+json"
```

Comment:
Consider making these constants so that changes to one variable can be applied everywhere.